### PR TITLE
Bugfix/token expired

### PR DESCRIPTION
**Pull request by nagkumar91**
# Description
The network calls to the RAI_SVC noticed a few calls failed with a message `Forbidden`. 
This should try to refresh the token while waiting for the response from RAI_SVC in exponential retry logic.

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_model_tools/_identity_manager.py [hunk 1]
```diff
@@ -121,6 +122,31 @@ def get_token(self) -> str:  # pylint: disable=invalid-overridden-method
             self.logger.info("Refreshed Azure endpoint token.")
 
         return self.token
+    
+    async def get_token_async(self) -> str:
```
Comment (luigiw) [inline sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_model_tools/_identity_manager.py:126]: Thx for the quick fix! 

Let's create a feature to clean up the logic here. If we need to refresh the token ourselves, we're prob not calling RAI service the right way. Ideally, we should use RAI service swagger to auto generate SDK client, which accepts token credential objects.

**Review by luigiw — APPROVED** [review]

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_model_tools/_identity_manager.py [hunk 2]
```diff
@@ -1,8 +1,10 @@
 # ---------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
+# pylint: disable=C0303
```
Comment (kdestin) [inline sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_model_tools/_identity_manager.py:4]:
```suggestion
```

Running a formatter should automatically resolve this

https://github.com/Azure/azure-sdk-for-python/blob/main/doc/eng_sys_checks.md#running-locally

**Review by kdestin — CHANGES_REQUESTED** [review]

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_model_tools/_identity_manager.py (hunk 2 above)
Comment (nagkumar91) [reply sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_model_tools/_identity_manager.py]: Updated :)

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_model_tools/_identity_manager.py [hunk 3]
```diff
@@ -87,6 +88,15 @@ def get_token(self) -> str:
...
         """
 
+    @abstractmethod
+    async def get_token_async(self) -> str:
+        """Async method to get the API token. Subclasses should implement this method.
+
+        :return: API token
+        :rtype: str
+        """
+        pass  # pylint: disable=unnecessary-pass
```
Comment (kdestin) [inline sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_model_tools/_identity_manager.py:98]:
```suggestion
```

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_model_tools/_identity_manager.py [hunk 4]
```diff
@@ -127,6 +137,31 @@ def get_token(self) -> str:
...
+        if (
+            self.token is None
+            or self.last_refresh_time is None
+            or time.time() - self.last_refresh_time > AZURE_TOKEN_REFRESH_INTERVAL
+        ):
+            self.last_refresh_time = time.time()
+            get_token_method = self.credential.get_token(self.token_scope.value)
+            if inspect.isawaitable(get_token_method):
+                # If it's awaitable, await it
+                token_response = await get_token_method
```
Comment (kdestin) [inline sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_model_tools/_identity_manager.py:155]:
```suggestion
                token_response: AccessToken = await get_token_method
```

Requires an import of `AccessToken` from `azure.core.credentials`

**Review by kdestin — CHANGES_REQUESTED** [review]

**Review by kdestin — APPROVED** [review]

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_model_tools/_identity_manager.py (hunk 1 above)
Comment (nagkumar91) [reply sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_model_tools/_identity_manager.py:139]: Created a task

**Pull request merged**